python/bind_data_array.h

```diff
@@ -0,0 +1,157 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
```

Comment:
Note that code in this file is just moved mostly unchanged, with exception of the `__setitem__` implementations, which should be reviewed more closely.

Reply:
Is the addition of `!is_bins(self[key])` the only change here?

Reply:
What is the reason for this change?
Does `self[key]` return a view of the coords / masks of the outer DataArray and therefore cannot assign something per event?

Reply:
Ok